apm-agent-core/src/main/java/co/elastic/apm/impl/transaction/AbstractSpan.java

```diff
@@ -138,12 +144,32 @@ public T activate() {
         final ElasticApmTracer tracer = this.tracer;
         previouslyActive = tracer.getActive();
         tracer.setActive(this);
+        List<SpanListener> spanListeners = tracer.getSpanListeners();
+        for (int i = 0; i < spanListeners.size(); i++) {
+            try {
+                spanListeners.get(i).onActivate(this);
+            } catch (Error e) {
```

Comment:
Isn't that the same as only catching Exceptions?

Reply:
I would like to only catch `Exception`s but `MethodHandle#invoke` throws `Throwable`